authhelper.go

```diff
@@ -0,0 +1,176 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/Pallinder/go-randomdata"
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/muir/nfigure"
+	"github.com/pkg/browser"
+)
+
+type Output struct {
+	ExpiresAt          string `json:"expire"` // RFC3339 format
+	Password           string `json:"password"`
+	Username           string `json:"username"`
+	Email              string `json:"email,omitempty"`
+	ModelVersionNumber int    `json:"modelVersionNumber"`
+}
+
+const baseURL = "https://singlestore.com/engine-sso"
+
+func main() {
+	main2(os.Stdout)
+}
+
+func main2(stdout io.Writer) {
+	var config struct {
+		Email     string   `flag:"email" validate:"omitempty,validate" help:"users SSO email address, if known"`
+		ClusterID []string `flag:"cluster-id,split=comma" help:"comma-separated list of specific clusters to access"`
+		Databases []string `flag:"databases,split=comma" help:"comma-separated list of specific databases to access"`
+	}
+	flagFiller := nfigure.PosixFlagHandler(nfigure.WithHelpText(""))
+	reg := nfigure.NewRegistry(nfigure.WithFiller("flag", flagFiller))
+	err := reg.Request(&config)
+	if err != nil {
+		panic(err.Error())
+	}
+	err = reg.Configure()
+	if err != nil {
+		panic(err.Error())
+	}
+
+	path := "/" + randomdata.Alphanumeric(20)
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	// Using httptest since it takes care of picking a random port
```

Comment:
since the browser session is going to be making a call to this local server, should we pick a static port number?  How will the browser know what port to make the call to?

Reply:
@mitchwadair 
I pass that to FE.  It's the `returnTo` parameter.  You must validate that the URL starts with `http://localhost:`